src/ImageSharp/Processing/Processors/Effects/PixelRowDelegateProcessor.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Effects
+{
+    /// <summary>
+    /// Applies a user defined row processing delegate to the image.
+    /// </summary>
+    public sealed class PixelRowDelegateProcessor : IImageProcessor
```

Comment:
The name may conflict with the future generic variant I'm proposing.
Instead of going deeper in the naming rabbit hole, I suggest internalizing it for now.

Reply:
I'm happy to internalize the processors for now.